feat: Add docker setup for IOTA-Names indexer

DaughterOfMars opened the pull request:
# Description of change

Adds a docker setup for the IOTA-Names indexer.

Closes #6999

DaughterOfMars committed:
feat: Add docker setup for IOTA-Names indexer

use dev profile

fix docker ignore

fix target dir

move prometheus file

fixes and add assets

Update docker/iota-names-indexer/build.sh

Co-authored-by: Thoralf-M <46689931+Thoralf-M@users.noreply.github.com>

remove version

Thoralf-M commented on dev-tools/iota-names-indexer/docker-compose.yml:
```diff
@@ -0,0 +1,49 @@
+        - PROFILE=dev
+    image: iota-names-indexer:dev
+    ports:
+      - "9100:9100/tcp" # Metrics
```
Currently this is set to `9184` in the code

DaughterOfMars replied:
fixed

Thoralf-M commented on docker/iota-names-indexer/build.sh:
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
+# Copyright (c) Mysten Labs, Inc.
+# Modifications Copyright (c) 2024 IOTA Stiftung
+# SPDX-License-Identifier: Apache-2.0
+./../utils/build-script.sh --image-tag "iotaledger/iota"
```
```suggestion
#!/bin/bash

# Copyright (c) 2025 IOTA Stiftung
# SPDX-License-Identifier: Apache-2.0

./../utils/build-script.sh --image-tag "iotaledger/iota"
```

thibault-martinez commented on dev-tools/iota-names-indexer/docker-compose.yml:
```diff
@@ -0,0 +1,53 @@
+version: "3"
```
```
WARN[0000] /Users/thibault/iota/iota/dev-tools/iota-names-indexer/docker-compose.yml: the attribute `version` is obsolete, it will be ignored, please remove it to avoid potential confusion 
```

muXxer reviewed (approved):
LGTM!

Can you maybe add a new entry in the CODEOWNERS as well?

Thoralf-M committed:
Merge branch 'feat/iota-names-indexer' into feat/add-docker-compose

thibault-martinez commented on dev-tools/iota-names-indexer/assets/prometheus/prometheus.yml:
Do we actually need this file?

DaughterOfMars replied:
Yes

DaughterOfMars committed:
fixes

DaughterOfMars committed:
Merge branch 'feat/add-docker-compose' of https://github.com/iotaledger/iota into feat/add-docker-compose

DaughterOfMars committed:
add network

DaughterOfMars committed:
entrypoint and port

Thoralf-M committed:
fix(iota-names-indexer): fix docker setup, change logs (#7046)

# Description of change

Fix docker setup so the services can work together, also changed and
added logs

## Type of change

- Bug fix (a non-breaking change which fixes an issue)
- Enhancement (a non-breaking change which adds functionality)

## How the change has been tested

`docker compose -f dev-tools/iota-names-indexer/docker-compose.yml up
--build`

thibault-martinez commented on docker/iota-names-indexer/build.sh:
```diff
@@ -0,0 +1,6 @@
+# Copyright (c) 2025 IOTA Stiftung
+# SPDX-License-Identifier: Apache-2.0
+
+./../utils/build-script.sh --image-tag "iotaledger/iota"
```
```suggestion
./../utils/build-script.sh --image-tag "iotaledger/iota-names-indexer"
```

thibault-martinez commented on dev-tools/iota-names-indexer/assets/grafana/datasources.yaml:
```diff
@@ -0,0 +1,15 @@
+apiVersion: 1
+
+datasources:
```
This makes it so we automatically have a data source from prometheus in grafana available?

DaughterOfMars replied:
Yeah basically

thibault-martinez commented on dev-tools/iota-names-indexer/docker-compose.yml:
```diff
@@ -0,0 +1,65 @@
+      args:
+        - GIT_REVISION
+        - BUILD_DATE
+        - PROFILE=dev
```
why dev?

DaughterOfMars replied:
apparently debug was renamed at some point

thibault-martinez replied:
Yeah sure, my question was rather: why not release?

DaughterOfMars replied:
Oh, haha. Well I figured for development we could leave it like this and then we might have a prod version of the Dockerfile and compose.

thibault-martinez replied:
Sounds like something we'll definitely forget to change but alright

thibault-martinez commented on docker/iota-names-indexer/Dockerfile:
```diff
@@ -0,0 +1,62 @@
+COPY crates crates
+COPY external-crates external-crates
+COPY iota-execution iota-execution
+COPY Cargo.toml Cargo.lock ./
```
I don't know docker that much, this copy syntax is weird, what do it do?

DaughterOfMars replied:
it is copying from the local directory to the one on the container

thibault-martinez replied:
And out of curiosity, why is it done differently than the previous copies?

DaughterOfMars replied:
I don't think it is, it's just copying multiple files

thibault-martinez replied:
I'll trust you that it works but these definitely seems like two different forms. First one looks like source/destination, second one is just sources

thibault-martinez commented on docker/iota-names-indexer/Dockerfile:
```diff
@@ -0,0 +1,62 @@
+WORKDIR "$WORKDIR"
+
+# Install runtime dependencies and tools
+RUN apt update && apt install -y libpq5 ca-certificates curl
```
Is the update necessary?

DaughterOfMars replied:
not sure. In this case I just took what the other examples do

DaughterOfMars committed:
fix build script

alexsporn commented on crates/iota-names-indexer/src/main.rs:
```diff
@@ -43,14 +49,12 @@ async fn main() -> Result<()> {
             PathBuf::from("./chk".to_string()), /* path to a local directory where checkpoints
                                                  * are stored. */
-            Some("http://localhost:9000/api/v1".to_string()),
+            Some("http://host.docker.internal:9000/api/v1".to_string()),
```
This should not be hardcoded to docker

thibault-martinez replied:
We are just making it work for now, we'll replace it when we have CLI args https://github.com/iotaledger/iota/issues/7018
